### PR TITLE
skeleton full consumer

### PR DESCRIPTION
**Pull request by dulalsaurab**

### CODE REVIEW

File: psync-fconsumer/full-consumer.h [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+    // Store sequence number for the prefix.
+    ndn_name_t prefix;
+    uint64_t sequence_number;
+} prefixes;
+
+typedef struct
+{
+// define all the required variables i.e. face, io-service and all
+
+    ndn_unix_face_t* unix_face;
```
Comment (RoySCU) [inline psync-fconsumer/full-consumer.h]: Thanks for your interest of contributing! But NDN-Lite doesn't have any "actual" faces but a base class and a dummy face for test use. 

If you want to use UNIX face, I wonder the [POSIX package](https://github.com/named-data-iot/ndn-iot-package-over-posix) is more appropriate for you to contribute to? Also  the `psync-fconsumer` looks like a consumer for PSync, which is a **protocol** and related files should be put into `app-support`.

Comment (yoursunny) [reply psync-fconsumer/full-consumer.h]: PSync protocol shouldn't depend on particular face type. It only needs a pointer to the base type.

File: psync-fconsumer/full-consumer.h [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+
+} full_consumer;
+
+// const ndn::Name& syncPrefix,
+//                    ndn::Face& face,
+//                    const UpdateCallback& onUpdate,
+//                    ndn::time::milliseconds syncInterestLifetime
+
+
+void init_full_consumer (char* syncPrefix, ndn_udp_face_t* face,
```
Comment (yoursunny) [inline psync-fconsumer/full-consumer.h]: const char*

File: psync-fconsumer/full-consumer.h [hunk 3]
```diff
@@ -0,0 +1,55 @@
...
+// const ndn::Name& syncPrefix,
+//                    ndn::Face& face,
+//                    const UpdateCallback& onUpdate,
+//                    ndn::time::milliseconds syncInterestLifetime
+
+
+void init_full_consumer (char* syncPrefix, ndn_udp_face_t* face, 
+                         void (*onUpdate)(void*));
+
+void send_sync_interet();
```
Comment (yoursunny) [inline psync-fconsumer/full-consumer.h]: Every function name should start with ndn_psync_full_consumer

File: psync-fconsumer/full-consumer.h [hunk 4]
```diff
@@ -0,0 +1,55 @@
...
+#include <stdlib.h>
+#include <ndn-lite.h>
+#include "ndn-lite/encode/name.h"
+#include "ndn-lite/encode/data.h"
+#include "ndn-lite/encode/interest.h"
+#include <time.h>
+#include <stdlib.h>
+
+
+#ifndef FULL_CONSUMER_H
```
Comment (yoursunny) [inline psync-fconsumer/full-consumer.h]: Include guard goes before #include lines

**Comment by dulalsaurab** [comment]
Sorry this was pushed by mistake, it's wip. We will have better version in
future.

Saurab,

File: psync-fconsumer/full-consumer.c [hunk 5]
```diff
@@ -0,0 +1,97 @@
+/*
```
Comment (Zhiyi-Zhang) [inline psync-fconsumer/full-consumer.c]: I would suggest you make psync as a module under /app-support, leave some APIs available to developers.
The consumer example can be placed under ndn-lite-Posix repo.

But I am fine with the current ongoing code, we can do the aforementioned things later when codes are ready.

**Comment by dulalsaurab** [comment]
WIP,  not ready for PR.

**Pull request closed without merging**